hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsStatistics.java

```diff
@@ -0,0 +1,280 @@
+    describe("Testing counter values got by deleting directory and files "
+        + "in Abfs");
+    AzureBlobFileSystem fs = getFileSystem();
+    /*
+     This directory path needs to be root for triggering the
+     directories_deleted counter.
+     */
+    Path createDirectoryPath = path("/");
+    Path createFilePath = path(getMethodName());
+
+    Map<String, Long> metricMap = fs.getInstrumentation().toMap();
+    //Test for initial statistic values before deleting any directory or files
+    assertEquals("Mismatch in op_delete", 0,
+        (long) metricMap.get("op_delete"));
+    assertEquals("Mismatch in files_deleted", 0,
+        (long) metricMap.get("files_deleted"));
+    assertEquals("Mismatch in directories_deleted", 0,
+        (long) metricMap.get("directories_deleted"));
+    assertEquals("Mismatch in op_list_status", 0,
+        (long) metricMap.get("op_list_status"));
+
+      /*
+      creating a directory and a file inside that directory.
+      The directory is root. Hence, no parent. This allows us to invoke
+      deleteRoot() method to see the population of directories_deleted and
+      files_deleted counters.
+       */
+    fs.mkdirs(createDirectoryPath);
+    fs.create(path(createDirectoryPath + getMethodName()));
+    fs.delete(createDirectoryPath, true);
+
+    metricMap = fs.getInstrumentation().toMap();
+
+      /*
+      Test for op_delete, files_deleted, op_list_status.
+      since directory is delete recursively op_delete is called 2 times.
+      1 file is deleted, 1 listStatus() call is made.
+       */
+    assertEquals("Mismatch in op_delete", 2,
+        (long) metricMap.get("op_delete"));
+    assertEquals("Mismatch in files_deleted", 1,
+        (long) metricMap.get("files_deleted"));
+    assertEquals("Mismatch in op_list_status", 1,
+        (long) metricMap.get("op_list_status"));
+
+      /*
+      creating a root directory and deleting it recursively to see if
+      directories_deleted is called or not.
+       */
+    fs.mkdirs(createDirectoryPath);
+    fs.create(createFilePath);
+    fs.delete(createDirectoryPath, true);
+    metricMap = fs.getInstrumentation().toMap();
+
+    // Test for directories_deleted.
+    assertEquals("Mismatch in directories_deleted", 1,
+        (long) metricMap.get("directories_deleted"));
+  }
+
+  /**
+   * Testing statistics of open, append, rename and exists method calls.
+   */
+  @Test
+  public void testOpenAppendRenameExists() throws IOException {
+    describe("Testing counter values on calling open, append and rename and "
+        + "exists methods on Abfs");
+
+    AzureBlobFileSystem fs = getFileSystem();
+    Path createFilePath = path(getMethodName());
+    Path destCreateFilePath = path(getMethodName() + "New");
+
+    Map<String, Long> metricMap = fs.getInstrumentation().toMap();
+
+    //Tests for initial values of counters before any method is called.
+    assertEquals("Mismatch in op_open", 0,
```

Comment:
Initial values of all counters are always zero. So we should create separate method to assert this rather than in each tests.